src/S3/S3Router.spec.ts

```diff
@@ -0,0 +1,47 @@
+import supertest from 'supertest'
+import { buildURL, createAuthHeaders } from '../../spec/utils'
+import { app } from '../server'
+import { getBucketURL } from './s3'
+
+const server = supertest(app.getApp())
+
+describe('S3 router', () => {
+  let fileName: string
+  let url: string
+
+  beforeEach(() => {
+    fileName = 'aFile'
+  })
+
+  describe('when getting the contents of a file', () => {
+    beforeEach(() => {
+      url = `/storage/contents/${fileName}`
+    })
+
+    it('should respond with a cached 301 redirecting to the S3 file', () => {
+      return server
+        .get(buildURL(url))
+        .set(createAuthHeaders('get', url))
+        .expect('Location', `${getBucketURL()}/contents/${fileName}`)
+        .expect('Cache-Control', 'public,max-age=31536000,immutable')
+        .expect(301)
+        .then(() => undefined)
+    })
+  })
+
+  describe('when getting the information of a file', () => {
+    beforeEach(() => {
+      url = `/storage/contents/${fileName}`
+    })
+
+    it('should respond with a cached 301 redirecting to the S3 file', () => {
+      return server
+        .get(buildURL(url))
+        .set(createAuthHeaders('head', url))
+        .expect('Location', `${getBucketURL()}/contents/${fileName}`)
+        .expect('Cache-Control', 'public,max-age=31536000,immutable')
+        .expect(301)
+        .then(() => undefined)
```

Comment:
Are these `then(() => undefined)` really necessary?

Reply:
A call to then is needed so the request gets executed (it's what they use to perform it after building it), but using await improves it a lot so I changed it for await.